storage/scylla_api.c

```diff
@@ -403,6 +400,73 @@ static status_t create_edge_table(CassSession* session) {
 exit:
   return ret;
 }
+
+status_t create_identity_table(CassSession* session, bool is_need_drop) {
+  status_t ret = SC_OK;
+  if (is_need_drop) {
+    if (execute_query(session, "DROP TABLE IF EXISTS identity;") != CASS_OK) {
+      ta_log_error("drop identity table fail\n");
+      ret = SC_STORAGE_CASSANDRA_QUREY_FAIL;
+      goto exit;
+    }
+  }
+  if (execute_query(session,
+                    "CREATE TABLE IF NOT EXISTS identity("
+                    "code uuid, hash blob, status tinyint, PRIMARY KEY (code));") != CASS_OK) {
+    ta_log_error("create identity table fail\n");
+    ret = SC_STORAGE_CASSANDRA_QUREY_FAIL;
+    goto exit;
+  }
+  if (execute_query(session, "CREATE INDEX IF NOT EXISTS ON identity(status);") != CASS_OK) {
+    ta_log_error("create identity table index fail\n");
+    ret = SC_STORAGE_CASSANDRA_QUREY_FAIL;
+    goto exit;
+  }
+exit:
+  return ret;
+}
+status_t init_identity(CassCluster** cluster, CassSession* session, char* hosts, bool is_need_drop,
+                       const char* keyspace_name) {
+  if (hosts == NULL) {
+    ta_log_error("%s\n", "SC_TA_NULL");
+    return SC_TA_NULL;
+  }
+  status_t ret = SC_OK;
+  CassStatement* use_statement = NULL;
+  char* use_query = NULL;
+  /* Add contact points */
+  *cluster = create_cluster(hosts);
+  if (connect_session(session, *cluster) != CASS_OK) {
+    ta_log_error("%s\n", "connect Scylla cluster fail");
+    ret = SC_STORAGE_CONNECT_FAIL;
+    goto exit;
+  }
+  if ((ret = create_keyspace(session, keyspace_name)) != SC_OK) {
+    ta_log_error("%s\n", "create permanent keyspace fail");
+    goto exit;
+  }
+  ret = make_query(&use_query, "USE ", keyspace_name, "");
+  if (ret != SC_OK) {
+    ta_log_error("%s\n", "make USE keyspace query fail");
+    goto exit;
+  }
+  use_statement = cass_statement_new(use_query, 0);
+  if (execute_statement(session, use_statement) != CASS_OK) {
+    ta_log_error("%s\n", "SC_STORAGE_CASSANDRA_QUREY_FAIL");
+    ret = SC_STORAGE_CASSANDRA_QUREY_FAIL;
+    goto exit;
+  }
+
+  if ((ret = create_identity_table(session, is_need_drop)) != SC_OK) {
+    ta_log_error("%s\n", "create identity table fail");
+    goto exit;
```

Comment:
This line is unnecessary

Reply:
I am afraid of someone adds code after this line forgets to add `goto exit;` back.